### PR TITLE
Sdk swaps pallet refactor first stage

### PR DESCRIPTION
**Pull request by morkeltry**
Separates out the functions which accept `Asset | object | string` as a parameter from those which strictly require `Asset`.
These wrappers are now all in a Swaps class property called `forgiving`.
Intention in future is to make the .forgiving functions the default and instead place the strict functions in a .strict object.

Tightens up the use of `AssetId`, now renamed `AssetShortform`, and retained only for flexibity of parameters to `forgiving.pooljoinMultiFunc`.

Made the use of default public class methods (`someMethod =. .. `) and ES6 function syntax consistent in Swaps class.

### CODE REVIEW

File: packages/sdk/src/util/index.ts [hunk 1]
```diff
@@ -33,7 +35,43 @@ const tolerantJsonParse = (anything) => {
...
+
+  if (!api) {
+    throw new Error(
+      "SDK must be initialised and an `api` passed in order to crate a reference type"
+    );
+  }
+  return api.createType("Asset", AssetShortformFromString(stringAsset));
+};
+
+const AssetShortformFromString = (
```
Comment (lsaether) [inline packages/sdk/src/util/index.ts:62]: camelCase

**Review by lsaether — COMMENTED** [review]
Overall looks good, but should camelCase functions.

File: packages/cli/src/actions/getSpotPrice.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
-import SDK, { util } from "@zeitgeistpm/sdk";
+// import SDK, { util } from "@zeitgeistpm/sdk";
+import SDK, { util } from "../../../sdk/src";
```
Comment (lsaether) [inline packages/cli/src/actions/getSpotPrice.ts:2]: This should be changed back to use package paths

File: packages/sdk/src/types/index.ts [hunk 3]
```diff
@@ -25,7 +25,7 @@ type categoricalOutcomeIndex = [number, number];
 
 type scalarOutcomeIndex = [number, "Long" | "Short"];
 
-export type AssetId =
+export type AssetShortform =
```
Comment (lsaether) [inline packages/sdk/src/types/index.ts:28]: I don't think this type needs to exist, as it's an incomplete implementation of `Asset` and doesn't not contain enough information to create a proper `CategoricalOutcome` or `ScalarOutcome`.

Comment (lsaether) [inline packages/sdk/src/types/index.ts:28]:
```suggestion
export type AssetJson =
```

Comment (morkeltry) [reply packages/sdk/src/types/index.ts:28]: create an issue

File: packages/cli/src/actions/getSpotPrice.ts (hunk 2 above)
Comment (morkeltry) [inline packages/cli/src/actions/getSpotPrice.ts:2]:
```suggestion
import SDK, { util } from "@zeitgeistpm/sdk";
```

**Pull request closed without merging**